cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go

```diff
@@ -175,59 +175,83 @@ func executeExchangeWrapperTests(ctx context.Context, t *testing.T, exch exchang
 	t.Helper()
 	iExchange := reflect.TypeOf(&exch).Elem()
 	actualExchange := reflect.ValueOf(exch)
+
+	var firstPriority []string
+	var secondPriority []string
+
 	for x := 0; x < iExchange.NumMethod(); x++ {
 		methodName := iExchange.Method(x).Name
 		if _, ok := excludedMethodNames[methodName]; ok {
 			continue
 		}
-		method := actualExchange.MethodByName(methodName)
-
-		var assetLen int
-		for y := 0; y < method.Type().NumIn(); y++ {
-			input := method.Type().In(y)
-			if input.AssignableTo(assetParam) ||
-				input.AssignableTo(orderSubmitParam) ||
-				input.AssignableTo(orderModifyParam) ||
-				input.AssignableTo(orderCancelParam) ||
-				input.AssignableTo(orderCancelsParam) ||
-				input.AssignableTo(pairKeySliceParam) ||
-				input.AssignableTo(getOrdersRequestParam) ||
-				input.AssignableTo(pairKeySliceParam) {
-				// this allows wrapper functions that support assets types
-				// to be tested with all supported assets
-				assetLen = len(assetParams) - 1
-			}
-		}
-		tt := time.Now()
-		e := time.Date(tt.Year(), tt.Month(), tt.Day()-1, 0, 0, 0, 0, time.UTC)
-		s := e.Add(-time.Hour * 24 * 2)
-		if methodName == "GetHistoricTrades" {
-			// limit trade history
-			e = time.Now()
-			s = e.Add(-time.Minute * 3)
+
+		if strings.Contains(methodName, "Update") {
```

Comment:
I think this would be better with a string slice of priority function names.
This way of doing it gives the appearance that `GetLastPairsUpdateTime` or `SupportsAutoPairUpdates`are important functions to run first. A string slice of function names is more customisable too.

```diff
diff --git a/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go b/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go
index fb42ba465..e26ba57fb 100644
--- a/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go
+++ b/cmd/exchange_wrapper_standards/exchange_wrapper_standards_test.go
@@ -184,10 +184,7 @@ func executeExchangeWrapperTests(ctx context.Context, t *testing.T, exch exchang
                if _, ok := excludedMethodNames[methodName]; ok {
                        continue
                }
-
-               if strings.Contains(methodName, "Update") {
-                       // Update functions are tested first as they are required for other
-                       // functions to work.
+               if _, ok := priorityMethodNames[methodName]; ok {
                        firstPriority = append(firstPriority, methodName)
                } else {
                        secondPriority = append(secondPriority, methodName)
@@ -567,6 +564,13 @@ type assetPair struct {
        Asset asset.Item
 }
 
+// priorityMethodNames are called before other exchange functions functions
+var priorityMethodNames = map[string]struct{}{
+       "UpdateTickers":     {}, // Is required before FetchTickers is called
+       "UpdateOrderbook":   {}, // Is required before FetchOrderbook is called
+       "UpdateAccountInfo": {}, // Is required before FetchAccountInfo is called
+}
+

```

Reply:
Nice 👍